src/new_currencies.jl

```diff
@@ -0,0 +1,61 @@
+abstract type Currency end
+
+struct Asset{C}
+    balance::Number
+    function Asset{C}(a::Number) where C <: Currency
+        C()
+        new{C}(a)
+    end
+end
+
+
+struct RegularCurrency{S} <: Currency
+    function RegularCurrency{S}() where {S}
+        haskey(_regular_data, S) || error("Regular currency $S is not defined.")
+        new{S}()
+    end
+end
+
+const _regular_data = Dict(
+	:DAI  => (RegularCurrency{:DAI},  18),
```

Comment:
What is 18? the total amount of the currency?

Reply:
It's the currency's precision. The number of allowed decimals after dot in the floating point representation of currency.
For doing arithmetic operations with the currency, we need this value.